src/controllers/portfolio/portfolio.ts

```diff
@@ -410,6 +420,35 @@ export class PortfolioController extends EventEmitter {
     }
   }
 
+  async updateFirstCashbackConfettiStatus({
+    accountId,
+    shouldShow,
+    toggleModal,
+    shouldGetAdditionalPortfolio
+  }: {
+    accountId: AccountId
+    shouldShow: boolean
+    toggleModal: boolean
+    shouldGetAdditionalPortfolio: boolean
+  }) {
+    if (toggleModal) {
+      this.#isFirstCashbackConfettiVisible = !this.#isFirstCashbackConfettiVisible
+    }
+
+    this.firstCashbackConfettiStatusByAccount = {
+      ...this.firstCashbackConfettiStatusByAccount,
+      [accountId]: shouldShow
+    }
```

Comment:
I'll discuss with you offline the option to tweak this a bit to something more scalable, not so niche (first cashback status) and not so UI-specific (confetti) like (pseudo-code):

```
cashbackStatusByAccount = {
  [accountAddr]: {
    firstCashbackReceivedAt: timestamp | null
    firstCashbackSeenAt: timestamp | null
  }
}
```